src/state.rs

```diff
@@ -267,11 +261,29 @@ impl ProxyState {
                 debug!("failed to fetch workload for {}", ep.workload_uid);
                 return None;
             };
+
+            let target_port = svc.ports.get(&addr.port());
             // If endpoint overrides the target port, use that instead
-            let target_port = ep.port.get(&addr.port()).unwrap_or(target_port);
+            let target_port = ep.port.get(&addr.port()).or(target_port);
+            let target_port = match (target_port, &wl.application_tunnel) {
```

Comment:
Based on the description of this PR:
```
The way this PR is written could possibly break things in the future though.. it would assume inbound_tunnel does port mapping. It sort of does that via application_tunnel.
```

Does it make sense to have a test to go with this so in the case that what you describe is actually an issue, it may be less subtle and easier to debug with a failing test to reference. (e.g. application tunnel eventually does something different than what we assume here about port mapping). Wdyt?

My main worry here is that this becomes one of those weird bugs that ppl spend a long time trying to figure out because it's not so obvious in the code.

Reply:
+1 weird behavior is fine as long as there's a unit test that describes it exactly.

Reply:
Test was added.